reader-gtfs/src/main/java/com/graphhopper/gtfs/PtLocationSnapper.java

```diff
@@ -47,11 +53,19 @@ public Result snapAll(List<GHLocation> locations, List<EdgeFilter> snapFilters)
         PointList points = new PointList(2, false);
         ArrayList<Snap> pointSnaps = new ArrayList<>();
         ArrayList<Supplier<Label.NodeId>> allSnaps = new ArrayList<>();
+        EncodingManager encodingManager =
+                graphHopperStorage.getEncodingManager();
+        final EnumEncodedValue<RoadClass> roadClassEnc = encodingManager.getEnumEncodedValue(RoadClass.KEY, RoadClass.class);
+        final EnumEncodedValue<RoadEnvironment> roadEnvEnc = encodingManager.getEnumEncodedValue(RoadEnvironment.KEY, RoadEnvironment.class);
         for (int i = 0; i < locations.size(); i++) {
             GHLocation location = locations.get(i);
             if (location instanceof GHPointLocation) {
                 GHPoint point = ((GHPointLocation) location).ghPoint;
-                final Snap closest = locationIndex.findClosest(point.lat, point.lon, snapFilters.get(i));
+                EdgeFilter mergedFilter = new SnapPreventionEdgeFilter(
+                        snapFilters.get(i), roadClassEnc, roadEnvEnc,
+                        Collections.singletonList("footway"));
```

Comment:
Would we still want to snap to sidewalks for public transit directions with walking as the connecting mode? Probably, right? If so, this should test for the connecting mode. Not currently relevant but matters when we add a transit/no bike directions mode or want to merge changes upstream.

Reply:
Makes sense. Let me try and implement this at the config-level, using a `snap_preventions` field underneath our `bike2` profile.

Reply:
Done, this is now handled via our configs. We can specify `snap_preventions` as a comma-separated list.